File: 10/10a.js
const fs = require('fs');

const input = fs.readFileSync('./10.txt', 'utf8');

const lines = input.split(/\r?\n/);

let X = 1;

let cycle = 1;

let res = 0;

for (const line of lines) {
  const parts = line.split(' ');
  if (cycle <= 220 && (cycle - 20) % 40 === 0) {
    const signalStrength = cycle * X;
    res += signalStrength;
  }

  cycle += 1;

  if (parts[0] === 'addx') {
    if (cycle <= 220 && (cycle - 20) % 40 === 0) {
      const signalStrength = cycle * X;
      res += signalStrength;
    }

    X += parseInt(parts[1], 10);
    cycle += 1;
  }
}

console.log(res);

File: 10/10a.py
with open('10.txt') as file:
    lines = file.read().split('\n')
    X = 1
    cycle = 1
    res = 0

    for line in lines:
        parts = line.split(' ')
        if cycle <= 220 and (cycle-20) % 40 == 0:
            signalStrength = cycle*X
            res += signalStrength
        cycle += 1

        if parts[0] == 'addx':
            if cycle <= 220 and (cycle-20) % 40 == 0:
                signalStrength = cycle*X
                res += signalStrength

            X += int(parts[1])
            cycle += 1

    print(res)
